File: Structural/Lab12/Typescript/src/estate_subsystem.ts
import { RealEstate } from "./Entities/real_estate";

export class EstateSubsystem {
    public estatePortfolio(estates:Array<RealEstate>):string{
        let sb = "";
        for (const estate of estates) {
            sb += `Estate ${estate.address}\n`;
        }

        return sb;
    }
}

File: Structural/Lab12/C#/EstateSubsystem.cs
using System;
using System.Collections.Generic;
using System.Text;
using C_.Entities;

namespace C_
{
    public class EstateSubsystem
    {
        public string EstatePortfolio(IEnumerable<RealEstate> realEstates)
        {
            var sb = new StringBuilder();
            foreach (var estate in realEstates)
            {
                sb.Append($"Estate {estate.Address}\n");
            }

            return sb.ToString();
        }
    }
}
